sphinx/tutorial/cylc/furthertopics/inheritance.rst

```diff
@@ -0,0 +1,603 @@
+
+.. spoiler:: Solution warning
+
+   .. code-block:: cylc
+
+         [[LAND_VEHICLE]]
+             inherit = VEHICLE
+             [[[meta]]]
+                 description = A vehicle which can travel over the ground
+
+         [[BICYCLE]]
+             inherit = LAND_VEHICLE
+             [[[meta]]]
+                 description = A small two-wheeled vehicle.
+
+         [[penny_farthing]]
+             inherit = BICYCLE, HUMAN_ENGINE
+             [[[meta]]]
+                 title = An old-fashioned bicycle.
+
+
+Using ``cylc get-config`` to inspect the configuration of the ``penny_farthing``
+task we can see that it inherits settings from the ``VEHICLE``,
+``BICYCLE`` and ``HUMAN_ENGINE`` families.
+
+.. code-block:: cylc
+
+   inherit = BICYCLE, HUMAN_ENGINE
+   init-script = echo 'Boarding'  # Inherited from VEHICLE
+   pre-script = echo 'Departing'  # Inherited from VEHICLE
+   post-script = echo 'Arriving'  # Inherited from VEHICLE
+   [[[environment]]]
+       FUEL = pizza               # Inherited from HUMAN_ENGINE
+   [[[meta]]]
+       description = A small two-wheeled vehicle.  # Inherited from LAND_VEHICLE - overwritten by BICYCLE
+       title = An old-fashioned bicycle.           # Defined in penny_farthing
+
+.. spoiler:: Hint hint
+
+   .. code-block:: bash
+
+      cylc get-config . --sparse -i "[runtime]penny_farthing"
+
+Hovercraft
+^^^^^^^^^^
+
+We will now add a hovercraft called the Hoverwork BHT130 better known to some as
+the Isle Of Wight Ferry.
+
+.. image:: https://upload.wikimedia.org/wikipedia/commons/e/e7/Hovercraft_leaving_Ryde.JPG
+   :width: 300px
+   :align: center
+   :alt: Hoverwork BHT130 Hovercraft
+
+Hovercraft can move over both land and water and in some respects can be thought
+of as flying vehicles.
+
+.. digraph:: Example
+   :align: center
+
+   size = "7,5"
+   bgcolor=none
+
+   VEHICLE [color=royalblue, fillcolor=powderblue, shape=box, style=filled]
+   AIR_VEHICLE [color=royalblue, fillcolor=powderblue, shape=box, style=filled]
+   LAND_VEHICLE [color=royalblue, fillcolor=powderblue, shape=box, style=filled]
+   WATER_VEHICLE [color=royalblue, fillcolor=powderblue, shape=box, style=filled]
+   HOVERCRAFT [color=royalblue, fillcolor=powderblue, shape=box, style=filled]
+   bht130 [color=royalblue, fillcolor=powderblue, shape=box, style=filled]
+   ENGINE [color=royalblue, fillcolor=powderblue, shape=box, style=filled]
+   INTERNAL_COMBUSTION_ENGINE [color=royalblue, fillcolor=powderblue, shape=box, style=filled]
+   VEHICLE -> AIR_VEHICLE -> HOVERCRAFT
+   VEHICLE -> LAND_VEHICLE -> HOVERCRAFT
+   VEHICLE -> WATER_VEHICLE -> HOVERCRAFT
+   HOVERCRAFT -> bht130
+   ENGINE -> INTERNAL_COMBUSTION_ENGINE -> bht130
+
+Write new families and one new task to represent the above structure.
+
+Add a description (``[meta]description``) to the ``WATER_VEHICLE`` and
+``HOVERCRAFT`` families and a title (``[meta]title``) to the ``bht130`` task.
+
+.. spoiler:: Solution warning
+
+   .. code-block:: cylc
```

Comment:
Once again, put full stops after descriptions for consistency.

Reply:
Done!